### ISSUE TITLE
can't load curl in Rstudio integrated R console, works in system terminal 

### ISSUE THREAD

**Issue by drammock** [rstudio/rstudio#8420]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.3.1093 
    OS Version      :  Ubuntu 20.04.1 LTS
    R Version       :  4.0.3

### Steps to reproduce the problem
1. create a conda environment (mine is called "sys" in the code samples/errors below)
2. install R and various R packages into that environment
3. Download the Rstudio `.deb` file and install in a terminal (using `dpkg`, with the relevant conda environment activated)
4. add `if(interactive()) tools:::httpdPort(10439)` to `~/.Rprofile` (otherwise when starting Rstudio you get `Error in tools::startDynamicHelp() : internet routines cannot be loaded` in the R console, and help system doesn't work)
5. add `export RSTUDIO_WHICH_R=/opt/miniconda3/envs/sys/bin/R` to `~/.profile`
6. log out & log back in, or reboot, just to make sure changes to `.profile` are picked up everywhere they might need to be
7. Open Rstudio and try to load the curl library, or, open Rstudio and try to download a file using, e.g., `readr::read_tsv(some_url)`

### Describe the problem in detail
Internet routines still cannot be loaded by the Rstudio built-in R console, so loading from URLs does not work. Works fine in a system terminal, and even works fine in Rstudio's terminal tab if I run R in there. Here is code to reproduce:

```r
iso_url <- "http://www-01.sil.org/iso639-3/iso-639-3.tab"
iso_table <- readr::read_tsv(iso_url, col_types=readr::cols(.default="c"))
```
resulting in this error:
```
! curl package not installed, falling back to using `url()`
Error in url(path) : internet routines cannot be loaded
In addition: Warning message:
In url(path) :
  unable to load shared object '/opt/miniconda3/envs/sys/lib/R/modules//internet.so':
  /opt/miniconda3/envs/sys/lib/R/modules//../.././libssl.so.1.1: undefined symbol: EVP_idea_cbc, version OPENSSL_1_1_0
```
Note that contrary to what the error message says, the curl package *is* installed:

```console
$ conda list | grep r-curl
r-curl                    4.3               r40hcdcec82_1    conda-forge
```

...but it can't be loaded: `library(curl)` yields this error:

```
Error: package or namespace load failed for ‘curl’ in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/opt/miniconda3/envs/sys/lib/R/library/curl/libs/curl.so':
  /opt/miniconda3/envs/sys/lib/R/library/curl/libs/../../../.././libssl.so.1.1: undefined symbol: EVP_idea_cbc, version OPENSSL_1_1_0
```

Note also that `internet.so` exists in the place it's being looked for:

```console
$ ls -al /opt/miniconda3/envs/sys/lib/R/modules/internet.so 
-rwxrwxr-x 2 drmccloy drmccloy 104968 Oct 10 05:56 /opt/miniconda3/envs/sys/lib/R/modules/internet.so*
```
Here are a few posts I've already read about `EVP_idea_cbc` being undefined: [one](https://stackoverflow.com/a/23025421/1664024) [two](https://stackoverflow.com/q/62592625/1664024) [three](https://askubuntu.com/q/1120951/92714).  The basic idea from all of them is that probably `libssl` and `libcrypto` are incompatible (e.g., maybe a conda-managed `libssl` is linking to a system `libcrypto` that was compiled with incompatible flags or something).  That seems not to be the case here; my conda-managed `internet.so` appears to be linked to a conda-managed `libssl` and `libcrypto` in the same conda environment, and `libssl` is in turn also correctly linked to that version of `libcrypto`:

```$console
$ ldd /opt/miniconda3/envs/sys/lib/R/modules/internet.so | grep 'ssl\|crypto'
	libssl.so.1.1 => /opt/miniconda3/envs/sys/lib/R/modules/../.././libssl.so.1.1 (0x00007f84928b3000)
	libcrypto.so.1.1 => /opt/miniconda3/envs/sys/lib/R/modules/../.././libcrypto.so.1.1 (0x00007f84925e7000)
	libk5crypto.so.3 => /opt/miniconda3/envs/sys/lib/R/modules/../.././libk5crypto.so.3 (0x00007f84924a0000)
$ ldd /opt/miniconda3/envs/sys/lib/libssl.so.1.1 | grep crypto
	libcrypto.so.1.1 => /opt/miniconda3/envs/sys/lib/./libcrypto.so.1.1 (0x00007f4146959000)
```

The same is true of libcurl:

```console
$ ldd /opt/miniconda3/envs/sys/lib/R/library/curl/libs/curl.so | grep 'ssl\|crypto'
  libssl.so.1.1 => /opt/miniconda3/envs/sys/lib/R/library/curl/libs/../../../.././libssl.so.1.1 (0x00007fa3b0d10000)
  libcrypto.so.1.1 => /opt/miniconda3/envs/sys/lib/R/library/curl/libs/../../../.././libcrypto.so.1.1 (0x00007fa3b0a44000)
  libk5crypto.so.3 => /opt/miniconda3/envs/sys/lib/R/library/curl/libs/../../../.././libk5crypto.so.3 (0x00007fa3b08fd000)
```

As mentioned above, the two-line code snippet using `read_tsv(some_url)` works in R in a system terminal, and also works in a terminal tab within Rstudio (if I run R within that terminal tab).  The only place it doesn't work is in the Rstudio integrated R console.  Prior to the above library link tracing, I had read somewhere that setting `LD_LIBRARY_PATH` correctly was the key to fixing such problems. So I (temporarily) added `export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/opt/miniconda3/envs/sys/lib` to `~/.profile`. After a reboot (just to make sure), the problem persisted. I have since removed that variable export from `~/.profile`, and although `$LD_LIBRARY_PATH` is undefined in a system terminal or the terminal tab of Rstudio (makes sense; they're not running `~/.profile`, they're running `~/.bashrc`), strangely in the Rstudio built-in R console, I get this:

```R
Sys.getenv("LD_LIBRARY_PATH")
[1] "/opt/miniconda3/envs/sys/lib/R/lib:"
```

I don't know if that is expected behavior or not; R from a system terminal or R from the Rstudio terminal tab both give `""` as the result of that command. Regardless, if I call `Sys.unsetenv("LD_LIBRARY_PATH")` and then try to load the URL again, it still doesn't work (same error about "internet routines", but without the additional warning about the undefined symbol).

**note**: this one conda env called "sys" is the only place on my system where R is installed. I've seen other issues/posts where a similar-on-the-surface error occurred because of multiple R installs. That is not what is going on here.

### Describe the behavior you expected

Loading the `curl` library should work in Rstudio's R console, just like it does in the system terminal or in Rstudio's terminal tab.

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [ ] ~~If I am reporting a RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).~~ n/a
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by drammock** [comment]
Crossref #7613. Apparently the solution that used to work there no longer works. (that solution was: write a custom shell script that initialized the "sys" conda env before launching Rstudio, so that `$PATH` would be correct, and alias `rstudio` to run that shell script).  I have no idea why that stopped working.

**Comment by ronblum** [comment]
@drammock Thank you for filing this issue—and wow, this is one of the most comprehensive tickets I've seen! I'm marking it as a bug to review as we continue developing the RStudio IDE.

**Comment by drammock** [comment]
Here's some more info that seems to me like it is probably useful.  I ran 

```console
$ conda activate sys
$ conda env config vars set LD_LIBRARY_PATH=/opt/miniconda3/envs/sys/lib
```
After deactivating and reactivating the `sys` environment so that the variable gets properly defined:

```console
$ rstudio
/bin/bash: /opt/miniconda3/envs/sys/lib/libtinfo.so.6: no version information available (required by /bin/bash)
(sys) [~] $ /usr/lib/rstudio/bin/crashpad_handler: /opt/miniconda3/envs/sys/lib/libcurl.so.4: no version information available (required by /usr/lib/rstudio/bin/crashpad_handler)
[2478972:2478972:20201118,134535.396250:ERROR process_memory_range.cc:86] read out of range
[2478972:2478972:20201118,134535.396280:ERROR elf_image_reader.cc:558] missing nul-terminator
[2478972:2478972:20201118,134535.396373:ERROR process_reader_linux.cc:445] no module mappings 0x7f41ff301000
[2478972:2478972:20201118,134535.397712:ERROR elf_dynamic_array_reader.h:61] tag not found
[2478972:2478972:20201118,134535.397853:ERROR elf_dynamic_array_reader.h:61] tag not found
[2478972:2478972:20201118,134535.398422:ERROR elf_dynamic_array_reader.h:61] tag not found
[2478972:2478972:20201118,134535.398453:ERROR elf_dynamic_array_reader.h:61] tag not found
[2478972:2478973:20201118,134535.419090:ERROR directory_reader_posix.cc:42] opendir: No such file or directory (2)
/home/drmccloy/.drammock/rstudio.sh: line 16: 2478969 Segmentation fault      (core dumped) /usr/lib/rstudio/bin/rstudio
```

**Comment by drammock** [comment]
This appears to be fixed in version 1.4.1103 (thanks!).  I even noticed that it's no longer necessary to add `if(interactive()) tools:::httpdPort(10439)` to `~/.Rprofile` to get the help pane to work.
